### PR TITLE
[MO] Updating condition in TF Const value extracting

### PR DESCRIPTION
**Pull request by yekruglov**
Root cause analysis: MO incorrectly stores TF Const with empty value, if it has ND-shape shape (N>1) 

Solution: Update condition in value extracting to avoid value returning before reshaping

Ticket: 57996

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests N/A: no new operations were enabled
* [x]  Transformation tests N/A: no new transformations were enabled
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list N/A: no new operations were enabled
* [x]  Guide on how to convert the **public** model N/A: no new models were enabled
* [x]  User guide update N/A

### CODE REVIEW

**Review by evolosen — APPROVED** [review]

**Review by iimironov — APPROVED** [review]

**Review by mvafin — APPROVED** [review]

File: model-optimizer/unit_tests/mo/front/tf/extractors/utils_test.py [hunk 1]
```diff
@@ -211,3 +211,17 @@ def test_str_decode_list(self):
         with self.assertLogs(log.getLogger(), level="ERROR") as cm:
             result = tf_tensor_content(pb_tensor.dtype, shape, pb_tensor)
             self.assertEqual([warning_message, warning_message], cm.output)
+
```
Comment (rkazants) [inline model-optimizer/unit_tests/mo/front/tf/extractors/utils_test.py:214]: it seems that we have no case with a scalar. For example, 
        pb_tensor = PB({
            'dtype': 3,
            'int_val': 4,
        })
Please add it.

File: model-optimizer/mo/front/tf/extractors/utils.py [hunk 2]
```diff
@@ -70,7 +70,12 @@ def tf_tensor_content(tf_dtype, shape, pb_tensor):
...
                 value = np.array(type_helper[1](pb_tensor))
 
-    if len(shape) == 0 or shape.prod() == 0:
+    # Ignore an empty value, if len(shape) > 1
+    # For example, value = [] and shape = [1, 1, 0]
+    # This is needed to reshape this value later and to return reshaped value = [[[]]]
+    # Otherwise there can be failures during partial inference, because we are storing an empty value with incorrect
+    # shape
+    if len(shape) == 0 or (len(shape) == 1 and shape.prod() == 0):
         if len(value) == 1:
```
Comment (rkazants) [inline model-optimizer/mo/front/tf/extractors/utils.py:79]: I can't understand about how it works for a scalar case. For example, `value=4` and `shape=[]`. Now it looks failing.
Please explain.

Comment (yekruglov) [reply model-optimizer/mo/front/tf/extractors/utils.py:79]: There is 'else` on line 86, scalar value will be returned as is

Comment (yekruglov) [reply model-optimizer/mo/front/tf/extractors/utils.py:79]: @rkazants Ok, now I see this problem

Comment (yekruglov) [reply model-optimizer/mo/front/tf/extractors/utils.py]: I propose to do something like this

**Comment by rkazants** [comment]
@yekruglov, please make CI green

**Review by rkazants — APPROVED** [review]

**Pull request merged**